# HG changeset patch
# User Rik <rik@octave.org>
# Date 1405621226 25200
#      Thu Jul 17 11:20:26 2014 -0700
# Node ID 961ecaf9fb57256fbf29a8991317dc88da9e7c3f
# Parent  4ff59f43f7ecdd919eaf0b9592b4d8e92552a4a6
legend.m: Tweak syntax when setting multiple "unit" properties to be more readable.

* legend.m (updatelegend): Tweak syntax when setting multiple "unit" properties
to be more readable.

diff --git a/scripts/plot/appearance/legend.m b/scripts/plot/appearance/legend.m
--- a/scripts/plot/appearance/legend.m
+++ b/scripts/plot/appearance/legend.m
@@ -1084,18 +1084,17 @@ function updatelegend (h, ~)
       switch (get (hax, "activepositionproperty"))
         case "position"
           set (hax, "outerposition", outerposition);
           set (hax, "position", position);
         case "outerposition"
           set (hax, "position", position);
           set (hax, "outerposition", outerposition);
       endswitch
-      ## Crazy syntax because set() only seems to support row vectors
-      set (hax, repmat ({"units"}, [1 numel(hax)]), units(:)');
+      set (hax, {"units"}, units);
 
       h = legend (hax(1), hplots, get (h, "string"));
     unwind_protect_cleanup
       recursive = false;
     end_unwind_protect
   endif
 
 endfunction
